### ISSUE TITLE
Add react 19 to peer deps

### ISSUE THREAD

**Issue by lucas-garrido** [AzureAD/microsoft-authentication-library-for-js#7455]
### Core Library

MSAL.js (@azure/msal-browser)

### Wrapper Library

MSAL React (@azure/msal-react)

### Public or Confidential Client?

Public

### Description

React 19 is now stable could you update the peer dependencies please ?

**Comment by arjenbloemsma** [comment]
I totally agree with @lucas-garrido. This is currently blocking our app from being upgraded.

**Comment by arjenbloemsma** [comment]
Created a PR to address this: #7467

**Comment by ritwickg** [comment]
This complete blocker for any react 19 upgrades with SSO

**Comment by arjenbloemsma** [comment]
Just mentioned this in the PR, but will also mention it here:

To unblock my project, I've published a npm package of the msal-react fork. This allows us to keep moving forward and use react 19 in dev and test while the official package has not yet been released. Once the new version of official msal-react is released we can simply change back to that package, rebuild and move on.

For those interested in following a similar approach; https://www.npmjs.com/package/@arjenbloemsma/msal-react

**Comment by bdbvb** [comment]
Another temporary workaround, from the consuming package.json file:

```
"overrides": {
    "@azure/msal-react": {
        "react": "^16.8.0 || ^17 || ^18 || ^19"
    }
}
```

**Comment by 3pleFly** [comment]
waiting for this..

**Comment by marvinjaworski** [comment]
Please prefer this problem, I have already tested it manually with the described fix and it seems to work very well. We are urgently waiting for the update to use react 19.

**Comment by julesblancphin** [comment]
Any updates on this?

**Comment by aclebson-sage** [comment]
> Any updates on this?

I resolved this issue by updating to the versions:
├── @azure/msal-browser@3.28.0
├── @azure/msal-react@2.2.0

**Comment by bdbvb** [comment]
> > Any updates on this?
> 
> I resolved this issue by updating to the versions: ├── @azure/msal-browser@3.28.0 ├── @azure/msal-react@2.2.0

FYI, this would be a downgrade/workaround, not really a resolution.

**Comment by aclebson-sage** [comment]
> > > Any updates on this?
> > 
> > 
> > I resolved this issue by updating to the versions: ├── @azure/msal-browser@3.28.0 ├── @azure/msal-react@2.2.0
> 
> FYI, this would be a downgrade/workaround, not really a resolution.

for this specific case, it was the solution since version constraints on other React packages

**Comment by greatzambino** [comment]
I also am blocked by this

**Comment by jansonjc** [comment]
> Another temporary workaround, from the consuming package.json file:
> 
> ```
> "overrides": {
>     "@azure/msal-react": {
>         "react": "^16.8.0 || ^17 || ^18 || ^19"
>     }
> }
> ```

Looks like a reasonable workaround. Did anyone get into any issues with it?

**Comment by greatzambino** [comment]
It seems to be working well on my end

**Comment by User6531** [comment]
> > Another temporary workaround, from the consuming package.json file:
> > ```
> > "overrides": {
> >     "@azure/msal-react": {
> >         "react": "^16.8.0 || ^17 || ^18 || ^19"
> >     }
> > }
> > ```
> 
> Looks like a reasonable workaround. Did anyone get into any issues with it?

A week in production, and no issues so far.

**Comment by tibble49** [comment]
Is there an ETA on this?

**Comment by benc-uk** [comment]
Guys come on, this is not cool. It's been ~5 months and no fix? 
If you were a one man team, I'd get it, but you ain't

**Comment by MarianoVeloso** [comment]
I'm not going to be sympathetic or anything like that. People pay to use Azure AD. RCs for React 19 have been available practically since last year, and it's been 5 months since React released version 19. It's disrespectful.

**Comment by julesblancphin** [comment]
@sameerag @peterzenz @jo-arroyo @tnorling please advise the community, there has been 0 communication for months now. As @MarianoVeloso mentioned this is a paid product and the lack of response and support is unprofessional.

**Comment by julesblancphin** [comment]
For anyone curious, there's been an update here https://github.com/AzureAD/microsoft-authentication-library-for-js/issues/7722#issuecomment-2836079014

Essentially React 19 support is not a priority for the MS team and there's some additional work required for the PRs that have been opened.

**Comment by timtucker-dte** [comment]
> I'm not going to be sympathetic or anything like that. People pay to use Azure AD. RCs for React 19 have been available practically since last year, and it's been 5 months since React released version 19. It's disrespectful.

There seems to be a broader issue across multiple MS projects / departments in supporting updated dependency / runtime versions.

A reasonable expectation from customers:

- Dependency releases a preview version
- Developers of components that build upon it (like MS) start making sure that their components are compatible
- Final version of dependency is released (~6 months after preview)
- Component developers release a version that supports the new dependency (~1 month after release)

What we see instead:

- Dependency releases a preview version
- MS is still playing catchup finalizing support for the previous version
- Final version of dependency is released
- Next preview version of the dependency is released
- MS releases a version that supports the new dependency (~6-12 months after release)

When combined with MS's support policy for dependencies, the net effect is that the N-2 version is often deprecated or out of support before N is supported.
